app/src/main/java/com/example/cinemapp/data/MovieRepository.kt

```diff
@@ -76,6 +105,31 @@ class MovieRepository(
         }
     }
 
+    suspend fun getMovieListAllPages(
+        movieListType: MovieListType,
+        sessionId: String? = null
+    ): List<MovieDTO> {
+        val firstPage = when (movieListType) {
+            MovieListType.UPCOMING -> remoteDataSource.getUpcoming()
+            MovieListType.POPULAR -> remoteDataSource.getPopular()
+            MovieListType.TOP_RATED -> remoteDataSource.getTopRated()
+            MovieListType.FAVORITE -> sessionId?.let { remoteDataSource.getFavorite(sessionId = it) }
+            MovieListType.WATCHLIST -> sessionId?.let { remoteDataSource.getWatchlist(sessionId = it) }
+            MovieListType.RATED -> sessionId?.let { remoteDataSource.getRated(sessionId = it) }
+        }?.let {
+            getBodyFromResponse(it)
+        }
+        val fullList = mutableListOf<MovieDTO>()
+        for (i in 1..(firstPage?.totalPages ?: 0)) {
```

Comment:
If `firstPage` is `null`, then the for loop will go from 1 to 0...will this crash the app? 😅

Reply:
The app will not crash, the for loop will perform the check to see if the current i (1) is smaller or equal to the final number and will just instantly exit the loop.